Update documentation

nagi-desuuu opened the pull request:
- Adds skinning info for newly-added elements.
- Modify existing documentation for grammar errors and miscellaneous corrections.
- Split off skinning sections into their own pages.
- More changes that are too long for this summary, so it's advised to read through the changes.

nagi-desuuu committed:
Update documentation

- Adds skinning info for newly-added elements.
- Modify existing documentation for grammar errors and miscellaneous corrections.
- Split off skinning sections into their own pages.
- More changes that are too long for this summary, so it's advised to read through the changes.

flustix commented on skinning/json.md:
```diff
@@ -0,0 +1,123 @@
+- `hit_position` decides how many pixels up from the bottom of the screen the hit position is.
+- `colors` tints the hit lighting of each individual column. The first entry will tint the lighting in the first lane from the left of the stage, the second entry will tint the second lane, and so on.
+
+This goes from 1K to 10K so all keymodes will need to be added.
```
1-3k and 9-10k only are accessible thru osu!mania maps. And it doesn't really *need* to be added.

flustix commented on skinning/json.md:
```diff
@@ -0,0 +1,123 @@
+
+The default colours for each judgement are as follows:
+
+|Judgement|Old Defaults|New Defaults|
```
"Old defaults"? They always have been the "New Defaults". The "old" ones are from the example skin and never actually have been in the game.

nagi-desuuu replied:
Should I use them in the template then?

flustix replied:
Maybe just remove the "Old" Column and rename "New Defaults" to "Color"

flustix commented on skinning/judgement.md:
```diff
@@ -0,0 +1,13 @@
+|Filename|Judgement|
+|---|---|
+|`flawless.png`|Maximum Hit Judgement|
+|`perfect.png`|300 Hit Judgement|
```
These numbers are from osu!. They have nothing to do with this.

nagi-desuuu replied:
Should I just copy this game's judgement names?

flustix replied:
Should be fine yeah

flustix commented on skinning/judgement.md:
```diff
@@ -0,0 +1,13 @@
+# Judgement
+These are the files that determine what image shows up for each of the judgements in-game. The files are labled as such:
```
Maybe mention that they should be in a folder. (Applies to all other sections too)

flustix commented on skinning/lighting.md:
```diff
@@ -0,0 +1,4 @@
+# Lighting
+This is what decides what image shows up when hitting each column. This is labelled as `column-lighting.png`, and is applied to all columns regardless of gamemode.
+
+It is possible to tint each column's lighting individually by following the skin.json commands.
```
Might be unclear. Maybe mention the "Colors" section.

flustix commented on skinning/samples.md:
```diff
@@ -0,0 +1,15 @@
+
+|Sample|Usage|
+|---|---|
+|`fail.wav`|Failing for any reason.|
```
Gameplay samples are in a subfolder. ("Samples/Gameplay")

flustix commented on skinning/stage.md:
```diff
@@ -0,0 +1,7 @@
+# Stage
+These decide how the non-playfield elements are rendered in game. These include `border-left.png` and `border-right.png` which decide how the left and right side of the playfield look like, and `hitline.png` which shows where the hitwindow is for notes.
```
background.png is missing here.

nagi-desuuu committed:
Tell users that 1-3K and 9-10K modes are optional

nagi-desuuu committed:
Add pointer about how each skin section should be in separate folders

nagi-desuuu committed:
Add link to json skinning doc in lighting.md

nagi-desuuu committed:
Add path to where gameplay samples should be stored

nagi-desuuu committed:
Add missing `background.png`

nagi-desuuu committed:
Add clearer info on how `foreground.png` could be cropped

I feel like I could do better, but I'll leave it as is for now.

nagi-desuuu committed:
Remove old defaults

Turns out, the old defaults were actually random colors from an example skin made not too long ago, and were never part of the actual game.

nagi-desuuu committed:
Rename Hit Judgements to match

Also included in this change is where exactly a user can change the colors for the judgements.
